File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head==null || head.next==null || k==0)return head;
        ListNode ptr=head;
        int n=1;
        while(ptr.next!=null){
            ptr=ptr.next;
            n++;
        }
        k=k%n;
        ptr.next=head;
        k=n-k;
        int count=0;
        while(k>0){
            ptr=ptr.next;
            k--;
        }
        head=ptr.next;
        ptr.next=null;
        return head;
    }
}

File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        int count=1;
        if(head==NULL || head->next==NULL)return head;
        ListNode* curr=head;
        while(curr->next){
            curr=curr->next;
            count++;
        }
        k=k%count;
        k=count-k;// do left rotation ehich will be similiar to right rotation
        curr->next=head;
        while(k>0){
            curr=curr->next;
            k--;
        }
        head=curr->next;
        curr->next=NULL;
        return head;

    }
};



